rapid7/ruby_smb
# Fix SMB1 Trans2 response packets

This PR fixes some SMB1 `Trans2` response packet structures that was causing error when reading responses:
- remove the `name` field in the `DataBlock`. This field only exists for `Trans2` requests, not for responses (see [SMB_COM_TRANSACTION2 - 2.2.4.46.2 Response](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-cifs/216e606a-eee1-4c3f-b88e-0eb14dc380b2)).
- remove `trans2_data` field and the preceding `pad` when the no data is sent by the message.

Also, this fixes an issue with SMB1 `File#rename` by removing the `utf-16le` encoding. Apparently, RubySMB does not support unicode for SMB1 packets.

## Verification
The `delete_file.rb` example script can be used to confirm this fixes the issue with SMB1.
1. Run `ruby examples/delete_file.rb 127.0.0.1 <username> <password> <share> <filename> 1`
2. **Verify** no error is returned
3. **Verify** the file has been deleted on the target

## Scenarios
- without these fixes:
```
$ ruby examples/delete_file.rb 127.0.0.1 msfuser msfuser temp test.txt 1
SMB1 : (0x00000000) STATUS_SUCCESS: The operation completed successfully.
Connected to \\127.0.0.1\temp successfully!
Traceback (most recent call last):
	1: from examples/delete_file.rb:40:in `<main>'
/home/msfuser/ruby_smb/lib/ruby_smb/smb1/file.rb:186:in `delete': Expecting SMB1 protocol with command=50, got SMB1 protocol with command=50, Status: (0x00000000) STATUS_SUCCESS: The operation completed successfully. (RubySMB::Error::InvalidPacket)
```
- with these fixes:
```
$ ruby examples/delete_file.rb 127.0.0.1 msfuser msfuser temp test.txt 1
SMB1 : (0x00000000) STATUS_SUCCESS: The operation completed successfully.
Connected to \\127.0.0.1\temp successfully!
(0x00000000) STATUS_SUCCESS: The operation completed successfully.
```
